playground/backend/internal/db/mapper/mapper.go

```diff
@@ -32,4 +32,5 @@ type ResponseMapper interface {
 	ToObjectInfo(exampleDTO *dto.ExampleDTO) *dto.ObjectInfo
 	ToDefaultPrecompiledObjects(defaultExamplesDTO *dto.DefaultExamplesDTO) map[pb.Sdk]*pb.PrecompiledObject
 	ToPrecompiledObj(exampleDTO *dto.ExampleDTO) *pb.PrecompiledObject
+	ToSnippetsDatasetsMap(datasets []*entity.DatasetEntity, datasetsSnippets []*entity.DatasetSnippetEntity) (map[string][]*dto.DatasetDTO, error)
```

Comment:
Why we don't keep it in Datastore as a map, by the way?
UPD: please, consider storing it in a `pg_snippets` entity